exams/pearson/readers.py

```diff
@@ -0,0 +1,100 @@
+"""
+Readers for data Pearson exports
+"""
+import csv
+from datetime import datetime
+
+import pytz
+
+from exams.pearson.constants import (
+    PEARSON_CSV_DIALECT,
+    PEARSON_DATETIME_FORMAT,
+)
+from exams.pearson.exceptions import InvalidTsvRowException
+
+
+class BaseTSVReader(object):
+    """
+    Base class for TSV file readers
+
+    It handles the high-level mapping and reading of the data.
+    Subclasses specify how the fields map.
+    """
+    def __init__(self, field_mappers, read_as_cls):
+        """
+        Initializes a new TSV writer
+
+        The first value of each fields tuple is the destination field name.
+        The second value is a str property path (e.g. "one.two.three") or
+        a callable that when passed a row returns a computed field value
+
+        Usage:
+
+            # maps column 'ABC' to field 'abc' and casts it to an int
+            # maps column 'DEF' to field 'def' and leaves it as a str
+            # Initializes each row into the namedtuple
+            Record = namedtuple('Record', ['abc', 'def'])
+
+            BaseTSVReader([
+                ('ABC', 'abc', int),
+                ('DEF', 'def'),
+            ], Record)
+
+        Arguments:
+            field_mappers (list(tuple)): list of tuple field mappers
+            read_as_cld (cls): class to instantiate row data
+        """
+        self.read_as_cls = read_as_cls
+        self.field_mappers = field_mappers
+
+    @classmethod
+    def parse_datetime(cls, dt):
+        """
+        Parses a datetime from Pearson's format
+        """
+        return datetime.strptime(dt, PEARSON_DATETIME_FORMAT).replace(tzinfo=pytz.UTC)
```

Comment:
Yes, I think that would be best to let the `ValueError` raise. If it doesn't parse that means Pearson made a breaking change to their API and I'd want it reported to sentry and to abort the processing.